runtime/tools/python/ttrt/common/api.py

```diff
@@ -968,6 +968,19 @@ def _execute(binaries):
                             for tensor in program.output_tensors:
                                 self.logging.debug(f"{tensor}\n")
 
+                            # TODO this is copied over from setup.py, should be imported from there instead.
+                            debug_runtime = os.environ.get("TT_RUNTIME_DEBUG", "OFF") == "ON"
+                            
+                            if debug_runtime:
+                                with open("runtime/tools/python/ttrt/utils/debug_tensors.py", "w") as f:
+                                    f.write("import numpy as np\n\n")
+
+                                    inputs = f"{program.input_tensors}".replace("tensor", "np.array")
+                                    f.write(f"device_input_tensors = {inputs}\n\n")
+
+                                    outputs = f"{program.output_tensors}".replace("tensor", "np.array")
+                                    f.write(f"device_output_tensors = {outputs}\n\n")
+
```

Comment:
ttrt has the ability to dump tensors to disk, @tapspatel is this still supported?

I'd much rather build a script around the dumped tensors than generating it here.

Reply:
you can still dump all tensors to disk if you provide --save-artifacts flag. This will save any artifact generated during runtime (including all the tensors) into ttrt-artifacts/<test_name> folder.

Reply:
@nsmithtt nah, I never meant for this to become a serious tool. We definitely need to design proper test infra.

I reckoned now while we fast iterate things, we need just a quick way to check if everything is ok. I needed it for https://github.com/tenstorrent/tt-mlir/pull/743 actually, that's why I wrote it. 

What happened to me was that I was looking at console output of `ttrt run` and the tensors were written in summarized form with `...` and I took first and last couple of numbers from input tensor, `np.exp`-ed them and I saw that only the first few match device output, not the last. Then I tinkered with `api.py` to remove summarization, and I noticed only first tile was good. Then I implemented tilizing of `np.array` just to see what is going on with other tiles. None of them except the first one was matched, which indicated a problem with my kernel and helped me catch it.

However you feel about it, not check it in, check it in for now, I just thought it might help.

Reply:
Also, the easiest way was just to produce a fake python file next to the script from which I can then easily `import` tensors, I know it's not proper way to do it.

Reply:
Actually, I might take https://github.com/tenstorrent/tt-mlir/issues/537 on myself if you agree, would be fun.

I propose we check this in for people to use for now, and I'll remove it as part of PR for fixing #537 . 

I don't know, suggest whatever you feel is the best.

Reply:
#537 would be awesome, and I agree I think it'll be an interesting task I was hoping to do it myself, but don't have time at least for the next couple weeks.

I think if you're free to pick up #537 next then let's not commit this PR.  Other folks can cherry-pick this PR if they want or just do visual golden inspection for now.